### PR TITLE
Adds multiple projects functionality

### LINKED ISSUE #46: Edit project idea
Right now, we can only have one project - We might need to refine this to allow for users to edit more than one

### PR DESCRIPTION
**Pull request by FlaIespa**
## Description
Closes #46 and #52 
This PR implements the **multiple project functionality** with several key updates and improvements:

- Added the ability to **create, view, and edit multiple projects**.
- Refactored and/or created the following components"
  - **`Card` Component**: Updated to handle dynamic button labels and interactions, such as "Edit Project".
  - **`MyProjectsContent` Component**: Displays a list of projects fetched dynamically from the backend and includes the "Add Project" functionality.
  - **`AddProjectCard` Component**: Added a separate reusable card for creating a new project.
- Introduced the **`project_idea` page**:
  - Populates form data dynamically based on the selected project.
  - Allows users to edit and update project details.
- Updated routing:
  - Integrated links for creating new projects (`/new_project_idea`).
  - Added functionality to navigate to "My Projects" (`/my_projects`) and the "Feed" page.
- Ensured mobile responsiveness for all cards, buttons, and pages.
- Adds the delete button with visibility only under the"My Projects Page" 

## Type of Change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Tests (adding or updating tests)
- [ ] Documentation update

### **Steps to Test:**

1. **My Projects Page**:
   - Navigate to `/my_projects`.
   - Verify that all projects are fetched and displayed correctly in cards.
   - Ensure that clicking **"Edit Project"** on a card redirects to `/project_idea` with the corresponding project details pre-populated in the form.
2. **Add New Project**:
   - On the `/my_projects` page, click the **"Add Project"** button.
   - Verify that it navigates to `/new_project_idea`.
3. **Project Idea Page**:
   - Test editing an existing project:
     - Confirm that form fields load with project data fetched from the backend.
     - Make edits and click "Save" to ensure updates are reflected in the database.
   - Test creating a new project:
     - Submit new project details and verify that the project is added to the database and visible on `/my_projects`.
4. **Home Page**:
   - Confirm the updated onboarding steps:
     - Step 1: Navigate to "Complete Your Profile" (`/personal-details`).
     - Step 2: Navigate to "Add a New Project" (`/new_project_idea`).
     - Step 3: Navigate to "Go to Your Projects" (`/my_projects`).
     - Step 4: Navigate to the "Feed" page (`/home`).

## Screenshots (if applicable)

Change in the root page 
Screenshot 2024-12-16 at 3 40 53 PM

Change in the Navbar 
Screenshot 2024-12-16 at 3 41 18 PM

Display of all my projects in the feed
Screenshot 2024-12-16 at 3 41 34 PM

Display of all my projects in the My Projects page
Screenshot 2024-12-16 at 3 42 43 PM

Clicking on the button "Add a Project" leads to the New Project Idea form where you add an idea
Screenshot 2024-12-16 at 3 43 16 PM

Delete Button addition
Screenshot 2024-12-16 at 4 32 51 PM

After clicking on the delete button, the project is deleted with appropriate snackbar
Screenshot 2024-12-16 at 4 36 34 PM

### CODE REVIEW

**Comment by avalucianelson** [comment]
Wow, Flávia! This is incredible! I especially love the little trash can -- so cute! 

Impressive implementation and huge leap forward for the product in terms of usability and alignment with our target audience. 

The only concern I might have is that I think Kyron mentioned anything we add during testing will be saved (so like, the project "hello," "test1", "project3"). Idk what we need to do to make sure those don't show up on the site, but I'm sure you and Kyron can coordinate to make that happen :)) 

Thanks so much for this contribution!

File: src/components/ProjectIdeaForm.tsx [hunk 1]
```diff
@@ -101,10 +113,10 @@ const ProjectIdeaForm: React.FC<ProjectIdeaFormProps> = ({
...
 		};
 
-		if (user) {
-			void fetchProjectDetails();
-		}
-	}, [user]);
+		fetchProjectDetails().catch((err) =>
+			console.error("Unexpected error:", err),
+		);
+	}, [user, projectId]);
```
Comment (lewiskyron) [inline src/components/ProjectIdeaForm.tsx:119]: Hi Fla,

Thank you for the thorough PR description detailing the steps for testing your new features. Additionally, you have adhered to our current component pattern when introducing new components.

For future enhancements, consider utilizing Next.js's prefetch method for navigation. Prefetching page data before redirecting can lead to faster navigation and an improved user experience.

Nonetheless, this has been really well done. Thank you again for the comprehensive PR description.
Kudos!

**Review by lewiskyron — APPROVED** [review]

File: src/components/ProjectIdeaForm.tsx (hunk 1 above)
Comment (lewiskyron) [reply src/components/ProjectIdeaForm.tsx:119]: I have tested this on the deployed preview and everything works as expected! Ready for merging.

**Comment by FlaIespa** [comment]
> Wow, Flávia! This is incredible! I especially love the little trash can -- so cute!
> 
> Impressive implementation and huge leap forward for the product in terms of usability and alignment with our target audience.
> 
> The only concern I might have is that I think Kyron mentioned anything we add during testing will be saved (so like, the project "hello," "test1", "project3"). Idk what we need to do to make sure those don't show up on the site, but I'm sure you and Kyron can coordinate to make that happen :))
> 
> Thanks so much for this contribution!

Thank you for your review, Ava! 
Yes, so you are right that adding a project in development would still show the project in production (the deployed website everyone has access to). However, after adding the projects to test the functionality, I removed them from the database (manually deleting them from Supabase when the deleting functionality was not working and using the delete functionality once it was implemented).

File: src/components/ProjectIdeaForm.tsx (hunk 1 above)
Comment (FlaIespa) [reply src/components/ProjectIdeaForm.tsx:119]: Thank you, Kyron! I appreciate the kudos :)  I'll take your suggestion for Next.js's prefetch method for next time. I'll go ahead and merge :)

**Pull request merged**